webpack/assets/javascripts/react_app/redux/API/APIHelpers.js

```diff
@@ -0,0 +1,60 @@
+import { STATUS } from '../../constants';
+
+const getModifiedState = (
+  key,
+  state,
+  newState,
+  payload,
+  manageResponseByID
+) => {
+  const { id } = payload;
+
+  if (manageResponseByID) {
+    if (state[key]) {
+      return state.set(key, {
+        ...state[key],
+        [id]: { ...state[key][id], ...newState },
+      });
+    }
+    return state.set(key, { ...state[key], [id]: newState });
+  }
+
+  return state.set(key, { ...state[key], ...newState });
+};
+
+export const handleRequest = (key, state, payload, manageResponseByID) => {
```

Comment:
With more than ~3 function arguments, it can start getting confusing when I call the function because the args don't have names.  For example, if I call the function like this: 
```js
handleRequest('POWER_STATUS', { ...storeState}, { ...payload}, true)
```
Just looking at this code doesn't tell me the purpose of all these values.  I prefer using destructuring to name the function arguments:
```suggestion
export const handleRequest = ({key, state, payload, manageResponseByID}) => {
```
This way there's no ambiguity about what each value is, and as an added bonus you get the destructured values in your function for free.